### ISSUE TITLE
JmmDNS binds loopback interface

### ISSUE THREAD

**Issue by dhufnagel** [jmdns/jmdns#69]
The JmmDNS binds on the loopback interface and advertises it's address on the network. This causes clients to receive a 127.0.0.1 or 0::1 ip address and failing to receive host.

I am not sure if it is on purpose to bind the loopback interface, but I really think it's address should not be advertised on other networks (like 192.168...). 

For my purpose I stopped binding loopback interfaces but I want some discussion before committing.

Kind regards,
dhufnagel

**Comment by kaikreuzer** [comment]
To me it makes sense what you say, but maybe somebody more familiar with the mDNS spec can comment as well?

**Comment by dhufnagel** [comment]
I further investigated this issue and here are some results and thoughts:

Linux does not even allow multicasts on the loopback device, so I think it is fine to ignore loopback devices in general. The only situation where it could be useful is if you have a Service and Client both on the same machine without network interfaces. I cannot think of many such situations.

I tried to find the culprit why the local loopback address is propagated in other networks. Therefor I skimmed the mDNS RFC. I think this issue comes from another issue with the Windows loopback device and IPv6.

I think that there is a bug in the Windows IPv6 implementation of the loopback device. If I multicast a link-local IPv6 packet to the loopback, it is also propagated to other ethernet devices. It does not with IPv4 link-local packets. So I also skimmed the IPv6 RFC where it's said: 

> Link-Local addresses are designed to be used for addressing on a single link for purposes such as auto-address configuration, neighbor discovery, or when no routers are present.Routers must not forward any packets with link-local source or destination addresses to other links.

IMHO this seems to be done wrong in Windows (10). As multicasts are not only propagated to the local ethernet interface but also through the network of the ethernet interface, a client gets the loopback IP address. This is whats causing me problems with iOS.

To sum it up: I suggest to not bind a JmDNS instance to a loopback device. So if you agree with it, I can create a pull-request of my changes.

**Comment by kaikreuzer** [comment]
Hi @dhufnagel, sorry for my late response. Yes, I am fine with your suggested change - please create a PR for it!

**Comment by kaikreuzer** [comment]
Actually, I just did, assuming this is the fix you wanted to propose: https://github.com/jmdns/jmdns/pull/87
